Hazelcast.Net/Hazelcast.IO.Serialization/PortableContext.cs

```diff
@@ -277,7 +277,7 @@ internal IClassDefinition Register(IClassDefinition cd)
             internal void SetClassVersion(int classId, int version)
             {
                 var hasAdded = _currentClassVersions.TryAdd(classId, version);
-                if (!hasAdded)
+                if (!hasAdded && _currentClassVersions[classId] != version)
                 {
                     throw new ArgumentException("Class-id: " + classId + " is already registered!");
```

Comment:
You may add these information to this exception message: the current version and the version.

Reply:
yes but I keep it same with error message on java client side